fossunited/chapters/doctype/foss_event_rsvp_submission/test_foss_event_rsvp_submission.py

```diff
@@ -52,48 +32,26 @@ def test_unpublish_on_max_count(self):
         rsvp = self.rsvp
 
         # When submission count reaches the max count
-        for i in range(int(rsvp.max_rsvp_count)):
-            frappe.get_doc(
-                {
-                    "doctype": RSVP_RESPONSE,
-                    "linked_rsvp": rsvp.name,
-                    "event": rsvp.event,
-                    "name1": fake.name(),
-                    "email": fake.email(),
-                    "im_a": "Student",
-                }
-            ).insert()
+        for _ in range(int(rsvp.max_rsvp_count)):
+            generate_rsvp_submission(linked_rsvp=self.rsvp.name)
 
         # Then the RSVP must be unpublished
         is_published = frappe.db.get_value(EVENT_RSVP, rsvp.name, "is_published")
         self.assertFalse(is_published)
 
     def test_add_to_email_group(self):
         # Given an RSVP form for an event
-        form = self.rsvp
-
-        frappe.set_user(WEBSITE_USER)
         # When an RSVP response is done by a user
-        frappe.get_doc(
-            {
-                "doctype": RSVP_RESPONSE,
-                "linked_rsvp": form.name,
-                "event": form.event,
-                "name1": fake.name(),
-                "email": WEBSITE_USER,
-                "im_a": "Student",
-            }
-        ).insert()
+        frappe.set_user(WEBSITE_USER)
+        generate_rsvp_submission(linked_rsvp=self.rsvp.name, email=WEBSITE_USER)
 
         # Then the email should be added to an email group linked to event for participants
         email_group = frappe.db.get_value(
-            "Email Group", {"event": form.event, "group_type": "Event Participants"}
+            "Email Group", {"event": self.rsvp.event, "group_type": "Event Participants"}
```

Comment:
nit - why not just `self.event` or `self.event.name` (whichever the right thing is)? why `self.rsvp.event`?